### PR TITLE
Changes for Displaying Archived workspaces

### PR DESCRIPTION
**Pull request by charan1312**

### CODE REVIEW

**Comment by jdamerow** [comment]
ok to test

**Comment by jdamerow** [comment]
image
back button should go back to project page not workbench

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java [hunk 1]
```diff
@@ -256,4 +257,22 @@ public ModelAndView unarchiveWorkspace(@Validated @ModelAttribute("workspaceform
 		return model;
 	}
 
+    @AccessPolicies({
+        @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 1, userRole = {
+                RoleNames.ROLE_COLLABORATOR_ADMIN,
+                RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN,
+                RoleNames.ROLE_PROJ_COLLABORATOR_CONTRIBUTOR }) })
+    @RequestMapping(value = "auth/workbench/{projectid}/showarchivedworkspace", method = RequestMethod.GET)
+    public String showArchivedWorkspaces(
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java]: this method should be in its own controller

Comment (charan1312) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java]: 1. I did the changes in relation to Show Inactive workspaces which is pointing to location.href='${pageContext.servletContext.contextPath}/auth/workbench/${project.projectid}'" 
2. The method for showing inactive workspace is in ActivateWSController.java file.

But I will make the changes as mentioned

**Comment by charan1312** [comment]
@jdamerow : story/QUAD-80-sprint2 Made the changes as per the comments

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListArchivedWSController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+import edu.asu.spring.quadriga.web.workspace.backing.ModifyWorkspace;
+import edu.asu.spring.quadriga.web.workspace.backing.ModifyWorkspaceFormManager;
+
+@Controller
+public class ListArchivedWSController {
+
+    @Autowired
+    ModifyWorkspaceFormManager workspaceFormManager;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListArchivedWSController.java]: should be private

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListArchivedWSController.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            Model model) throws QuadrigaStorageException,
+            QuadrigaAccessException {
+        List<ModifyWorkspace> archivedWSList = workspaceFormManager
+                .getArchivedWorkspaceList(projectId, principal.getName());
+        model.addAttribute("archivedWSList", archivedWSList);
+        model.addAttribute("listArchivedWSProjectid", projectId);
+
+        return "auth/workbench/workspace/showArchivedWorkspace";
+        
+
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListArchivedWSController.java]: remove empty lines

**Comment by jdamerow** [comment]
is that just for QUAD-80 or is this pull request for another ticket as well?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListArchivedWSController.java (hunk 2 above)
Comment (charan1312) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListArchivedWSController.java]: I modified it to private

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListArchivedWSController.java (hunk 3 above)
Comment (charan1312) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListArchivedWSController.java]: Removed blank lines

**Comment by charan1312** [comment]
Removed empty lines and added private qualifier

**Comment by charan1312** [comment]
The pull request is only for QUAD-80

**Comment by jdamerow** [comment]
ok to test

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListArchivedWSController.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import edu.asu.spring.quadriga.aspects.annotations.CheckedElementType;
+import edu.asu.spring.quadriga.aspects.annotations.ElementAccessPolicy;
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+import edu.asu.spring.quadriga.web.workspace.backing.ModifyWorkspace;
+import edu.asu.spring.quadriga.web.workspace.backing.ModifyWorkspaceFormManager;
+
+@Controller
+public class ListArchivedWSController {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ListArchivedWSController.java:23]: this controller and its methods need javadoc

**Comment by jdamerow** [comment]
I'll merge the pull request but please add javadoc an make a new pull request.

**Comment by jdamerow** [comment]
you can add the javadoc to the QUAD-80 story branch

**Pull request merged**